File: java-pi-pantry/src/test/java/com/langrsoft/pi/pantry/AShelfLifeRepository.java
package com.langrsoft.pi.pantry;

import org.junit.Test;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.assertThat;

public class AShelfLifeRepository {
    ShelfLifeRepository repo = new ShelfLifeRepository();

    @Test
    public void containsExistingValuesForKnownCategories() {
        assertThat(repo.contains("water"), is(true));
    }

    @Test
    public void allowsStoringShelfLifeInfoForNewCategories() {
        int refrigeratedDaysOfLife = 42;
        repo.add("goop", new ShelfLife(refrigeratedDaysOfLife, 0));

        assertThat(repo.get("goop").getRefrigerated(), is(equalTo(refrigeratedDaysOfLife)));
    }
}

File: csharp-pi-pantry/csharp-pi-pantry-test/DomainTests/AShelfLifeRepository.cs
using NUnit.Framework;
using Pipantry.Domain;

namespace Pipantry.Domain.Tests
{
    [TestFixture]
    public class AShelfLifeRepository
    {
        ShelfLifeRepository repo = new ShelfLifeRepository();

        [Test]
        public void containsExistingValuesForKnownCategories()
        {
            Assert.That(repo.contains("water"), Is.True);
        }

        [Test]
        public void allowsStoringShelfLifeInfoForNewCategories()
        {
            int refrigeratedDaysOfLife = 42;
            repo.add("goop", new ShelfLife(refrigeratedDaysOfLife, 0));

            Assert.That(repo.get("goop").getRefrigerated(), Is.EqualTo(refrigeratedDaysOfLife));
        }
    }
}
